src/erlazure_insights.erl

```diff
@@ -0,0 +1,130 @@
+-module(erlazure_insights).
+
+-export([
+        get_events/3,
+        get_time_offset_as_timestamp/2,
+        get_events_all/3,
+        follow_token/3,
+        create_time_bounded_filter/2
```

Comment:
minor thing/advice: maybe `construct_time_bounded_filter` is better name?

this one does not actually **create** anything (in terms of side effects)